handlers/external/handler_test.go

```diff
@@ -267,11 +268,36 @@ var customTestCases = []IncomingTestCase{
 	},
 }
 
+var extReceiveTestCases = []IncomingTestCase{
+	{
+		Label:                "Receive Valid Message",
+		URL:                  receiveURL + "?sender=%2B2349067554729&text=Join",
+		Data:                 "empty",
+		ExpectedRespStatus:   200,
+		ExpectedBodyContains: "Accepted",
+		ExpectedMsgText:      Sp("Join"),
+		ExpectedURN:          "ext:+2349067554729",
+	},
+	{
+		Label:                "Receive Valid Message trim spaces",
+		URL:                  receiveURL + "?sender=+2349067554729&text=Join",
+		Data:                 "empty",
+		ExpectedRespStatus:   200,
+		ExpectedBodyContains: "Accepted",
+		ExpectedMsgText:      Sp("Join"),
+		ExpectedURN:          "ext:2349067554729",
+	},
+}
+
 func TestIncoming(t *testing.T) {
 	RunIncomingTestCases(t, testChannels, newHandler(), handleTestCases)
 	RunIncomingTestCases(t, testSOAPReceiveChannels, newHandler(), handleSOAPReceiveTestCases)
 	RunIncomingTestCases(t, gmChannels, newHandler(), gmTestCases)
 	RunIncomingTestCases(t, customChannels, newHandler(), customTestCases)
+
+	extChannel := test.NewMockChannel("8eb23e93-5ecb-45ba-b726-3b064e0c56ab", "EX", "2020", "GM", nil)
+	extChannel.SetScheme(urns.External.Prefix)
```

Comment:
I found this trick we already used in 
https://github.com/nyaruka/courier/blob/36bcdaa526dc44806481cc669380aadbda75b0e7/handlers/twiml/handlers_test.go#L1141
better instead of modifying `test.NewMockChannel` method to pass schemes

Reply:
Fine for this PR but really `NewMockChannel` should take schemes, we should set that in all tests, and we try to make our tests as realistic as possible